src/ophyd_async/core/signal.py

```diff
@@ -63,29 +70,68 @@ def __init__(
     ) -> None:
         self._timeout = timeout
         self._initial_backend = self._backend = backend
+
         super().__init__(name)
 
+    def children(self) -> Iterator[Tuple[str, "Device"]]:
+        yield from ()
+
     async def connect(
         self,
         mock=False,
         timeout=DEFAULT_TIMEOUT,
         force_reconnect: bool = False,
         backend: Optional[SignalBackend[T]] = None,
     ):
-        if backend:
-            if self._initial_backend and backend is not self._initial_backend:
+        """
+        Exclude conditions:
+            - Switching between mock=True and mock=False (and vice versa)
+                across successive connect calls.
+            - Passing a different backend than the one initially set
+                or used in the previous successful connection.
+
+        Parameters
+        ----------
+        mock : bool, optional
+            uses the `MockSignalBackend`, by default False
+            If the backend is supplied then this must always be false.
+        timeout : float, optional
+            Time to wait before failing with a TimeoutError, by default DEFAULT_TIMEOUT
+        force_reconnect : bool, optional
+            Always force reconnection, by default False
+        backend: SignalBackend, optional
+            The backend to use, by default None.
+            If the device initalization has no backend, then this is required.
+            This value cannot be changed between connect() calls.
+            It is either the exact same object, or it fails.
+            User mainly for mocks.
```

Comment:
It's not really used for mocks.

```suggestion
            If the device was initialized without a backend then this is required.
            Cannot change value between connect() calls.
```